Shareableviz Docs for Azure and GCP

ravi-kumar-pilla opened the pull request:
## Description

* Provides documentation for updated shareableviz support to Azure and GCP

## Development notes

* Added steps for users to follow Azure, GCP Deployment
* Added notes for the changes in the shareableviz implementation in the upcoming release

## QA notes

* Users should be able to follow the docs and deploy Kedro-Viz on AWS, Azure and GCP

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

NeroOkwa commented on docs/source/share_kedro_viz.md:
```diff
@@ -23,20 +23,17 @@ When prompted for a project name, you can enter anything, but we will assume `Sp
-    1. You can automate the process of publishing and sharing your Kedro-Viz. This section describes the steps for AWS, which is the only cloud provider supported for automation at present. Integration with other cloud providers, namely Azure and GCP, will be added soon. 
-    2. In the absence of automated publish and share for other platforms, there is a [manual, platform-agnostic publish and share process](#platform-agnostic-sharing-with-kedro-viz) described below. You can use the manual process for sharing on static website hosts like GitHub pages, and cloud providers like Azure and GCP.
-
+1. You can automate the process of publishing and sharing your Kedro-Viz. This section describes the steps for AWS, which is the only cloud provider supported for automation at present. Integration with other cloud providers, namely Azure and GCP, will be added soon.
```
Do we need to update this since we are adding Azure and GCP?

ravi-kumar-pilla replied:
I added a note for 7.2.0 separately as I was not sure how to describe for users using viz sharing before 7.2.0

rashidakanchwala commented:
I would have the layout of this as 
Publish and Share Kedro-viz
 - Platform-agnostic Publish and share 
 - Publish and share automatically via cloud providers 
   - Publish and share via AWS 
   - Publish and share via Azure Blob Storage 
   - Publish and share via GCP 
   
 
 i would keep platform-agnostic via `kedro viz build` as the default way of sharing but if you want to deploy automatically, then we support the following cloud platforms
 
 What do you think @ravi-kumar-pilla  , @NeroOkwa , @astrojuanlu , @merelcht

NeroOkwa commented:
> I would have the layout of this page as :
> 
> * Publish and Share Kedro-viz
>   
>   * Platform-agnostic publish and share
>   * Publish and share automatically via cloud providers
>     
>     * Publish and share via AWS
>     * Publish and share via Azure Blob Storage
>     * Publish and share via GCS
> 
> i would keep platform-agnostic via `kedro viz build` as the default way of sharing but if you want to deploy automatically, then we support the following cloud platforms

@rashidakanchwala I agree with this, much better information architecture.

merelcht commented:
Agreed with the layout proposed by @rashidakanchwala. I'd also suggest adding a introduction section or page (a bit similar to https://docs.kedro.org/en/stable/notebooks_and_ipython/index.html) where you describe both approaches in short and link to the relevant sections. That way users don't have to scroll all the way through the page to find out what ways of publishing Viz exist.

rashidakanchwala commented on docs/source/share_kedro_vizNext.md:
```diff
@@ -0,0 +1,321 @@
+# Publish and share Kedro-Viz
```
can we break these into multiple pages, it is too much content in a single page.

rashidakanchwala commented on docs/source/share_kedro_vizNext.md:
```diff
@@ -0,0 +1,321 @@
+
+You can run the commands below for the required cloud-specific dependencies:
+
+AWS -
```
I was hoping the setup for each should be in the specific cloud provider pages

rashidakanchwala commented:
I suggest creating separate pages for each Publish and Share option:

- An introductory page for Publish and Share that presents and discusses the two ways of sharing. GitPages can be incorporated later, so we can omit it for now.
This page will then link to:
- Platform-agnostic Publish and Share (on its own page).
- Three separate pages dedicated for Publish and share with each cloud provider i.e. AWS, Azure, and GCP.

I believe consolidating all options onto one page could lead to confusion. As a user, when opting for a particular cloud provider, I would prefer straightforward steps tailored to that provider. This structure will provide clarity and ease of navigation.

ravi-kumar-pilla commented:
Hi @NeroOkwa , I tried fixing the build issue and addressed PR comments from Rashida. I moved the content around and created new files. I made few modifications to the content but most of it is intact. Please have a look and let me know if there is any issue. Thank you

rashidakanchwala commented on docs/source/share_kedro_viz.md:
````diff
@@ -21,116 +21,64 @@ kedro new --starter=spaceflights-pandas
-```bash
-export AWS_ACCESS_KEY_ID="your_access_key_id"
-export AWS_SECRET_ACCESS_KEY="your_secret_access_key"
+```{note}
````
can this not be after the entire point 2.

rashidakanchwala commented on docs/source/share_kedro_viz.md:
```diff
@@ -21,116 +21,64 @@ kedro new --starter=spaceflights-pandas
+1. **Platform-agnostic sharing with Kedro-Viz:** Using the kedro-viz build command, users can publish Kedro-Viz to any static website hosting platform such as GitHub pages, and share the URL generated. This is described in [Platform-agnostic sharing with Kedro-Viz](./platform_agnostic_sharing_with_kedro_viz)
 
-Select the user, then select `Create access key`. Follow the steps and create your keys.
+2. **Publish and share Kedro-Viz automatically:** You can automate the process of publishing and sharing your Kedro-Viz through AWS. Once the storage account and credentials are configured, users can then generate a shareable link from the Kedro-Viz UI using the 'Publish and share' button.
```
```suggestion
2. **Publish and share Kedro-Viz automatically:** You can automate the publishing and sharing of your Kedro-Viz through three different cloud providers: AWS, Azure, and GCS. Detailed instructions for each provider are outlined in the sections below.
```

rashidakanchwala commented on docs/source/share_kedro_viz.md:
````diff
@@ -21,116 +21,64 @@ kedro new --starter=spaceflights-pandas
+* [Publish and share Kedro-Viz on GCP](./publish_and_share_kedro_viz_on_gcp)
 
-You're now ready to publish and share your Kedro-Viz project. Start Kedro-Viz by running the following command in your terminal:
+```{toctree}
````
this should be hidden.

rashidakanchwala replied:
```{toctree}
:maxdepth: 1
:hidden:
platform_agnostic_sharing_with_kedro_viz
publish_and_share_kedro_viz_on_aws
publish_and_share_kedro_viz_on_azure
publish_and_share_kedro_viz_on_gcp
```

rashidakanchwala replied:
```suggestion
```{toctree}
:maxdepth: 1
:hidden:
platform_agnostic_sharing_with_kedro_viz
publish_and_share_kedro_viz_on_aws
publish_and_share_kedro_viz_on_azure
publish_and_share_kedro_viz_on_gcp
```
```

rashidakanchwala commented on docs/source/share_kedro_viz.md:
````diff
@@ -21,116 +21,64 @@ kedro new --starter=spaceflights-pandas
 
-You're now ready to publish and share your Kedro-Viz project. Start Kedro-Viz by running the following command in your terminal:
+```{toctree}
+:caption: Learn about Publish and Share Kedro-Viz
````
all of this should be hidden since we mention it above.

rashidakanchwala commented on docs/source/share_kedro_viz.md:
```diff
@@ -21,116 +21,64 @@ kedro new --starter=spaceflights-pandas
-Here's an example of the flow:
-
-![](./images/kedro-publish-share.gif)
+## Publish and share via CLI
```
Can this be also part of the cloud provider page.

ravi-kumar-pilla replied:
yes even I was in two minds about this. Since CLI command is common to all cloud providers, I mentioned it here. But even for running the CLI command, the setup needs to be done specific to cloud provider. May be we can move this to cloud provider specific page

rashidakanchwala commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
```diff
@@ -0,0 +1,105 @@
+
+## Configure your Azure Blob Storage and set credentials
+
+You can host your Kedro-Viz project on AzureBlobStorage. You must first create an Azure Storage account and then enable static website hosting. To do so, follow the [Azure tutorial](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-static-website-how-to?tabs=azure-portal) to configure a static website on AzureBlobStorage.
```
Can all these steps be in a numbered point format please?

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
```diff
@@ -0,0 +1,121 @@
+## Publish and share via CLI
+
+
+From Kedro-Viz version 8.0.0, the `kedro viz deploy` command takes **platform**, **endpoint** and **bucket name** as its options. You can use the following command from the root folder of your Kedro project
```
before 8.0.0 we do not have support for Azure. So, may be we can mention that at the top of this page , gcp and remove this statement

NeroOkwa replied:
Resolved.

ravi-kumar-pilla commented on docs/source/share_kedro_viz.md:
````diff
@@ -21,116 +21,55 @@ kedro new --starter=spaceflights-pandas
-export AWS_ACCESS_KEY_ID="your_access_key_id"
-export AWS_SECRET_ACCESS_KEY="your_secret_access_key"
+```{note}
+From Kedro-Viz version 8.0.0, in addition to AWS, you will be able to publish and share your Kedro-Viz on Azure and GCP.
````
May be we can remove this note from here and keep it in cloud provider specific pages like for Azure - From Kedro-Viz version 8.0.0, you will be able to publish and share your Kedro-Viz on Azure.

NeroOkwa replied:
Resolved.

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
```diff
@@ -0,0 +1,121 @@
+
+## Step 2: Configure your Azure Blob Storage
+
+You can host your Kedro-Viz project on Azure Blob Storage. You must first create an Azure Storage account and then enable static website hosting. To do so, follow the [Azure tutorial](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-static-website-how-to?tabs=azure-portal) to configure a static website on AzureBlobStorage.
```
Can we number the steps inside the step `Configure your Azure Blob Storage` ?

NeroOkwa replied:
Resolved.

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
```diff
@@ -0,0 +1,121 @@
+You pay for storing objects on your AzureBlobStorage. The amount you pay depends on the volume of data stored per month, quantity and types of operations performed, along with any data transfer costs, data redundancy option selected.
+
+See the official [Azure documentation](https://azure.microsoft.com/en-us/pricing/details/storage/blobs/) for more information.
+
```
It would be nice to have `### Permissions and access control` and `### Billing` section before `## Step 3: Set credentials` and repeat the same for other cloud providers

NeroOkwa replied:
I've kept `### Permissions and access control` before `## Set credentials` and left `### Billing` were it is.

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
````diff
@@ -0,0 +1,121 @@
+```{important}
+Having a `$web` container in your AzureBlobStorage is mandatory to use Kedro-Viz publish and share feature on Azure. For more information, see the official Azure documentation about 
+[Setting up a static website](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-static-website#setting-up-a-static-website).
+```
````
Can we move the above `{note}` and `{important}` sections before `## Step 3: Set credentials` ?

NeroOkwa replied:
Resolved.

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
````diff
@@ -0,0 +1,121 @@
+Click the **Publish and share** icon in the lower-left of the application. You will see a modal dialog to select your relevant AWS Bucket Region and enter your Bucket Name.
+
+```{note}
+From Kedro-Viz version 8.0.0, you will see a modal dialog to select your hosting platform, input your bucket name and endpoint link. The endpoint link can be found under **Storage account -> Capabilities -> Static website -> Primary endpoint**.
````
if we choose to mention that azure support is added from 8.0.0 at the top of this page, we can modify the note to be - `You will see a modal dialog to select your hosting platform, input your bucket name and endpoint link. The endpoint link can be found under **Storage account -> Capabilities -> Static website -> Primary endpoint**.` without **From Kedro-Viz version 8.0.0,**

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
````diff
@@ -0,0 +1,121 @@
+
+```{note}
+We will be updating the user flow doc for v8.0.0 soon...
+```
````
we can just have the v8.0.0 flow for azure and gcp as there was no support before. But for AWS, we can have both the flows

NeroOkwa replied:
Resolved.

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
```diff
@@ -0,0 +1,104 @@
+
+You pay for storing objects on your Google Cloud Storage. The amount you pay depends on the amount of data stored, data processing and network usage. Additionally you may be charged for using Cloud Load Balancing.
+
+See the official [Google Cloud Storage Billing](https://cloud.google.com/storage/pricing) and [Google Cloud Load Balancing Billing](https://cloud.google.com/vpc/network-pricing#lb) for more information.
```
same comment as Azure

NeroOkwa replied:
I decided to leave the billing section where it is, but updated the others.

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
```diff
@@ -0,0 +1,98 @@
+
+You pay for storing objects in your S3 buckets. The amount you pay depends on your objects’ size, how long you stored the object during the month, and the storage class.
+
+See the official [AWS documentation](https://aws.amazon.com/s3/pricing/?nc=sn&loc=4) for more information.
```
same comment as Azure

NeroOkwa replied:
Resolved.

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
```diff
@@ -0,0 +1,98 @@
+You can host your Kedro-Viz project on Amazon S3. You must first create an S3 bucket and then enable static website hosting. 
+
+To do so, follow the [AWS tutorial](https://docs.aws.amazon.com/AmazonS3/latest/userguide/HostingWebsiteOnS3Setup.html) to configure a static website on Amazon S3.
+
```
can we mention the note here - 
```{note}
Uploading your site's files will be done through Kedro-Viz
```

NeroOkwa replied:
Resolved.

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
````diff
@@ -44,55 +56,34 @@ export AWS_SECRET_ACCESS_KEY="your_secret_access_key"
 
 ```{note}
-From Kedro-Viz version 8.0.0, you will see a modal dialog to select your hosting platform, input your bucket name and endpoint link. The endpoint link can be found under **S3 bucket -> properties -> Static website hosting -> Bucket website endpoint**.
+You will see a modal dialog to select your hosting platform, input your bucket name and endpoint link. The endpoint link can be found under **S3 bucket -> properties -> Static website hosting -> Bucket website endpoint**.
````
since aws sharing had a different modal before 8.0.0, it might be worth mentioning, **From Kedro-Viz version 8.0.0** here

rashidakanchwala commented on docs/source/share_kedro_viz.md:
```diff
@@ -21,116 +21,56 @@ kedro new --starter=spaceflights-pandas
 
 When prompted for a project name, you can enter anything, but we will assume `Spaceflights` throughout.
```
I think this whole section shouldn't be a part of the intro page anymore. But each tutorial page should start off with this section similar to how the 'Preview Datasets' docs look

This page introduces the feature and different ways to use it  - https://kedro--1757.org.readthedocs.build/projects/kedro-viz/en/1757/preview_datasets.html
This page contains the guide including how to set up spaceflights project - https://kedro--1757.org.readthedocs.build/projects/kedro-viz/en/1757/preview_matplotlib_datasets.html

NeroOkwa replied:
@rashidakanchwala done!

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
```diff
@@ -0,0 +1,118 @@
+
+Step 11: Once those details are complete, click **Publish**. A hosted, shareable URL will be returned to you after the process completes.
+
+Here is an example of the flow (TODO - Need to add flows specific to cloud provider):
```
we can remove this TODO text

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
```diff
@@ -0,0 +1,140 @@
+
+Once those details are complete, click **Publish**. A hosted, shareable URL will be returned to you after the process completes.
+
+Here is an example of the flow (TODO - Need to add flows specific to cloud provider):
```
we can remove this TODO text and update the new flow

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
```diff
@@ -0,0 +1,120 @@
+
+Step 6: Once those details are complete, click **Publish**. A hosted, shareable URL will be returned to you after the process completes.
+
+Here is an example of the flow (TODO - Need to add flows specific to cloud provider):
```
we can remove this TODO text and add new flow

ravi-kumar-pilla commented on docs/source/share_kedro_viz.md:
```diff
@@ -1,136 +1,46 @@
+Kedro users want to share their pipeline visualisation with other non-technical  stakeholders on their team. 
 
-## Publish and share Kedro-Viz automatically
+Kedro enables users to do this by publishing Kedro-Viz to a static website hosting platform to share it with others. This guide uses the spaceflights tutorial as an example.
```
I feel the intro is a bit more like 2 statements (line 3,5) which are not combined properly here. I am not able to think of anything right now but I feel the intro can be improved. any ideas @rashidakanchwala

merelcht commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
```diff
@@ -0,0 +1,118 @@
+# Publish and Share via AWS
```
See guidance on capitalisation: https://github.com/kedro-org/kedro/wiki/Kedro-documentation-style-guide#capitalisation

merelcht commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
````diff
@@ -0,0 +1,118 @@
+When your project is ready, navigate to the root directory of the project. Install the dependencies from the project root directory by typing the following in your terminal:
+
+```bash
+pip install -r src/requirements.txt
````
```suggestion
pip install -r requirements.txt
```

merelcht commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
```diff
@@ -0,0 +1,120 @@
+# Publish and Share via GCP
```
```suggestion
# Publish and share via GCP
```

merelcht commented on docs/source/share_kedro_viz.md:
```diff
@@ -1,136 +1,46 @@
-When prompted for a project name, you can enter anything, but we will assume `Spaceflights` throughout.
-
-When your project is ready, navigate to the root directory of the project.
+# Publish and Share Kedro-Viz
```
```suggestion
# Publish and share Kedro-Viz
```

merelcht commented on docs/source/share_kedro_viz.md:
```diff
@@ -1,136 +1,46 @@
-## Publish and share Kedro-Viz automatically
+Kedro enables users to do this by publishing Kedro-Viz to a static website hosting platform to share it with others. This guide uses the spaceflights tutorial as an example.
 
+There are two ways to publish and share Kedro-Viz:
```
I found it a bit confusing that this says there are two ways of sharing Viz and then below there's also the header "Publish and share via CLI", so is this a third way or is it part of one of the other two?

merelcht commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
```diff
@@ -0,0 +1,118 @@
+For more information, see the official AWS documentation about [how to work with credentials](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html).
+
+
+### Billing
```
Why have this section on Billing? That has nothing to do with Kedro-Viz or Kedro, so I don't really see why it should be added.

merelcht commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
```diff
@@ -0,0 +1,118 @@
+
+## Publish and share the project
+
+Step 9: Once your Cloud storage is configured and the credentials are set, you are now ready to publish and share your Kedro-Viz project. Start Kedro-Viz by running the following command in your terminal:
```
```suggestion
Step 9: Once your cloud storage is configured and the credentials are set, you are now ready to publish and share your Kedro-Viz project. Start Kedro-Viz by running the following command in your terminal:
```

merelcht commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
````diff
@@ -0,0 +1,118 @@
+kedro viz run
+```
+
+Step 10: Click the **Publish and share** icon in the lower-left of the application. You will see a modal dialog to select your relevant AWS Bucket Region and enter your Bucket Name.
````
```suggestion
Step 10: Click the **Publish and share** icon in the lower-left of the application. You will see a modal dialog to select your relevant AWS Bucket Region and enter your bucket Name.
```

merelcht commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
````diff
@@ -0,0 +1,118 @@
+![](./images/kedro-publish-share.gif)
+
+```{note}
+We will be updating the user flow doc for v8.0.0 soon...
````
I wouldn't add this, it seems a bit sloppy and the user can't do anything with this information.

merelcht commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
```diff
@@ -0,0 +1,140 @@
+
+![](./images/azure_app_registration.png)
+
+When registration finishes, the app registration's Overview pane is displayed. You see the Application (Client) ID and Directory (Tenant) ID. We will use these values to set our environment variables.
```
```suggestion
When registration finishes, the app registration's overview pane is displayed. You see the Application (Client) ID and Directory (Tenant) ID. We will use these values to set our environment variables.
```

merelcht commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
```diff
@@ -0,0 +1,140 @@
+For more information, see the official Azure documentation about [how to work with environmental credentials](https://learn.microsoft.com/en-us/dotnet/api/azure.identity.environmentcredential?view=azure-dotnet).
+
+
+### Billing
```
Same comment as for AWS on the Billing section.

merelcht commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
```diff
@@ -0,0 +1,140 @@
+
+## Publish and share the project
+
+Once your Cloud storage is configured and the credentials are set, you are now ready to publish and share your Kedro-Viz project. 
```
```suggestion
Once your cloud storage is configured and the credentials are set, you are now ready to publish and share your Kedro-Viz project. 
```

merelcht reviewed (commented):
I've left some comments. Some things I saw several times:

- Capitalisation should be following these guidelines: https://github.com/kedro-org/kedro/wiki/Kedro-documentation-style-guide#capitalisation
- There was a bug in Kedro `0.19.0` so if possible it is better to recommend users to use `0.19.1` instead
- The added images aren't super clear in the final result when I look at https://kedro--1748.org.readthedocs.build/projects/kedro-viz/en/1748/publish_and_share_kedro_viz_on_aws.html
- I don't really understand why we're adding sections on "Billing" for each of the providers. It has nothing to do with Kedro or the feature.

NeroOkwa commented on docs/source/share_kedro_viz.md:
```diff
@@ -21,116 +21,56 @@ kedro new --starter=spaceflights-pandas
+There are two ways to publish and share Kedro-Viz:
 
-Add the IAM user to the user group (this is only possible if the group has been created).
+1. **Platform-agnostic sharing with Kedro-Viz:** Using the kedro-viz build command, users can publish Kedro-Viz to any static website hosting platform such as GitHub pages, and share the URL generated. This is described in [Platform-agnostic sharing with Kedro-Viz](./platform_agnostic_sharing_with_kedro_viz)
```
Thanks @merelcht for the review.

- Capitalisation was used for IAM user as that is the unique name used by AWS (https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users.html). We've retained it so as not to confuse users setting up their S3 bucket.

- The section on 'Billing' was added as it was a common user question and barrier  to adopting shareable Viz highlighted during user testing: https://github.com/kedro-org/kedro-viz/issues/1533

rashidakanchwala commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
````diff
@@ -0,0 +1,140 @@
+kedro viz run
+```
+
+Step 10: Click the **Publish and share** icon in the lower-left of the application. You will see a modal dialog to select your relevant AWS Bucket Region and enter your Bucket Name.
````
This is inaccurate.

rashidakanchwala commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
````diff
@@ -0,0 +1,120 @@
+kedro viz run
+```
+
+Step 5: Click the **Publish and share** icon in the lower-left of the application. You will see a modal dialog to select your relevant AWS Bucket Region and enter your Bucket Name.
````
This too is inaccurate.

rashidakanchwala commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
```diff
@@ -0,0 +1,120 @@
+# Publish and Share via GCP
+Kedro-Viz version 8.0.0 introduces publish and share via GCP.
+
+## Install Kedro
```
It should be 'Setup kedro project' or something like that

ravi-kumar-pilla commented on demo-project/data/session_store.db:
we should not be pushing this I guess

ravi-kumar-pilla commented on docs/source/conf.py:
```diff
@@ -18,15 +18,15 @@
 # The short X.Y version.
-version = re.match(r"^([0-9]+\.[0-9]+).*", release).group(1)
+version = re.match(r"^([0-9]+\.[0-9]+).*", "7.1.0").group(1)
 
```
we should revert this change

rashidakanchwala replied:
yeah, i have no idea what this was for.

merelcht reviewed (approved):
I think the structure is much clearer now and each of the pages is very good to follow 👏

stephkaiser commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
```diff
@@ -0,0 +1,115 @@
+# Publish and share via GCP
+
+This page describes how to publish Kedro-Viz to Azure to share it with others. It uses the spaceflights tutorial as an example.
```
small typo here - instead of Azure it should say GCP for the 'Publish and share via GCP' page

ravi-kumar-pilla replied:
how to publish Kedro-Viz **on** GCP

stephkaiser reviewed (commented):
Only one minor comment on the Azure/GCP typo. I also just remembered about versioning/updates on the generated shareable links and think it would be nice to also include a section about this so its clear to the user that the shared link does not get updated and highlight the timestamp in the top right corner (we've had questions from users about this before) but I don't want to block this release so maybe we can think about how to include this in the next version. Overall, this is great though and reads much better! its super clear and easy to follow as well, well done!

ravi-kumar-pilla commented on docs/source/share_kedro_viz.md:
```diff
@@ -1,136 +1,21 @@
+The publish and share feature on Kedro-Viz enables seamless sharing of pipeline visualizations on any static website hosting platform. There are two methods:
 
-Create a user group from the IAM dashboard, ensuring the user group has full access to the AWS S3 policy.
+1. **Platform-agnostic sharing with Kedro-Viz:** Use the `kedro viz build` command to publish Kedro-Viz to platforms like GitHub pages, sharing the generated URL. This is described in [Platform-agnostic sharing with Kedro-Viz](./platform_agnostic_sharing_with_kedro_viz)
```
[Nit] Use the `kedro viz build` command to publish Kedro-Viz **on** platforms

ravi-kumar-pilla commented on docs/source/platform_agnostic_sharing_with_kedro_viz.md:
````diff
@@ -0,0 +1,47 @@
+pip install -r requirements.txt
+```
+
+3. Kedro-Viz requires specific minimum versions of `fsspec[s3]`, and `kedro` to publish your project. Ensure you have these correct versions by updating the `requirements.txt` file of the Kedro project to add the following:
````
Since we separated s3fs dependency from 8.0.0, we only need minimum versions of fsspec, kedro and not fsspec[s3]

ravi-kumar-pilla commented on docs/source/platform_agnostic_sharing_with_kedro_viz.md:
```diff
@@ -0,0 +1,47 @@
+
+Introduced in Kedro-Viz version 7.1.0, the `kedro viz build` command allows seamless publishing and sharing of Kedro-Viz on any static website hosting platform. Running this command from the CLI generates a `build` folder within your Kedro project, containing a static Kedro-Viz app package.
+
+This page describes how to publish Kedro-Viz to a static website hosting platform using `kedro viz build`. It uses the spaceflights tutorial as an example.
```
[Nit] how to publish Kedro-Viz **on** a static website hosting...

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
```diff
@@ -0,0 +1,119 @@
+# Publish and share via AWS
+
+This page describes how to publish Kedro-Viz to AWS to share it with others. It uses the spaceflights tutorial as an example.
```
[Nit] how to publish Kedro-Viz **on** AWS

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
````diff
@@ -0,0 +1,119 @@
+pip install -r requirements.txt
+```
+
+Kedro-Viz requires specific minimum versions of `fsspec[s3]`, and `kedro` to publish your project. Ensure you have these correct versions by updating the `requirements.txt` file of the Kedro project to add the following:
````
Since we separated s3fs dependency from 8.0.0, we only need minimum versions of fsspec, kedro and not fsspec[s3]

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
````diff
@@ -0,0 +1,119 @@
+---------------
+
+```{important}
+The endpoint link can be found under **S3 bucket -> properties -> Static website hosting -> Bucket website endpoint**.
````
**S3 bucket -> Properties -> Static website hosting -> Bucket website endpoint**

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
```diff
@@ -0,0 +1,119 @@
+
+### Publish and share via CLI
+
+Use the `kedro viz deploy` command to publish Kedro-viz on AWS. You can execute the following command from your project's root folder:
```
[Nit] Use the `kedro viz deploy` command to publish Kedro-**V**iz on AWS.

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
````diff
@@ -0,0 +1,119 @@
+kedro viz deploy --platform=aws --endpoint=[s3-endpoint] --bucket-name=[s3-bucket-name]
+```
+
+If you are on Kedro-viz 7.0.0 you can still publish and share Kedro-Viz project using the existing command 
````
[Nit] If you are on Kedro-**V**iz 7.0.0

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
```diff
@@ -0,0 +1,128 @@
+# Publish and share via Azure
+
+This page describes how to publish Kedro-Viz to Azure to share it with others. It uses the spaceflights tutorial as an example.
```
[Nit] how to publish Kedro-Viz **on** Azure

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
````diff
@@ -0,0 +1,128 @@
+pip install -r requirements.txt
+```
+
+Kedro-Viz requires specific minimum versions of `fsspec[s3]`, and `kedro` to publish your project. Ensure you have these correct versions by updating the `requirements.txt` file of the Kedro project to add the following:
````
Since we separated s3fs dependency from 8.0.0, we only need minimum versions of fsspec, kedro and not fsspec[s3]

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
```diff
@@ -0,0 +1,128 @@
+You can host your Kedro-Viz project on Azure Blob Storage. 
+
+1. Enable static website hosting - Follow the [tutorial](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blob-static-website-how-to?tabs=azure-portal) to configure static website hosting on Azure Blob Storage. Please note that uploading your site's files will be done through Kedro-Viz
+
```
if we have the below note, we can remove - `Please note that uploading your site's files will be done through Kedro-Viz` statement from above

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_azure.md:
```diff
@@ -0,0 +1,128 @@
+
+### Publish and share via CLI
+
+Use the `kedro viz deploy` command to publish Kedro-viz on Azure. You can execute the following command from your project's root folder:
```
[Nit] Use the `kedro viz deploy` command to publish Kedro-**V**iz on Azure.

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
````diff
@@ -0,0 +1,115 @@
+pip install -r requirements.txt
+```
+
+Kedro-Viz requires specific minimum versions of `fsspec[s3]`, and `kedro` to publish your project. Ensure you have these correct versions by updating the `requirements.txt` file of the Kedro project to add the following:
````
Since we separated s3fs dependency from 8.0.0, we only need minimum versions of fsspec, kedro and not fsspec[s3]

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
```diff
@@ -0,0 +1,115 @@
+
+2. Ensure the `Compute Engine API` is enabled for your project as mentioned in the tutorial.
+
+3. Set up load balancer and SSL certificate: If serving your website through HTTPS, [set up a load balancer](https://cloud.google.com/storage/docs/hosting-static-website#lb-ssl) and configure an SSL certificate. 
```
If they need to serve it via http or https, they need a load balancer. However for HTTPS, they need an SSL cert

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
```diff
@@ -0,0 +1,115 @@
+
+3. Set up load balancer and SSL certificate: If serving your website through HTTPS, [set up a load balancer](https://cloud.google.com/storage/docs/hosting-static-website#lb-ssl) and configure an SSL certificate. 
+
+4. Set environment variable: For file uploads through Kedro-Viz, set `GOOGLE_APPLICATION_CREDENTIALS` as an environment variable. Follow these steps:
```
This section can be made similar to Azure. Since we are not setting env variable here, we can say **Obtain Application Credentials** or something similar to Azure where we say **Obtain parameters**. We can remove the statement - For file uploads through Kedro-Viz, set GOOGLE_APPLICATION_CREDENTIALS as an environment variable

Step4 can be broken down into - Create Service account and assign a role, Generate Service Account Key and download it

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
```diff
@@ -0,0 +1,115 @@
+
+### Publish and share via CLI
+
+Use the `kedro viz deploy` command to publish Kedro-viz on Azure. You can execute the following command from your project's root folder:
```
Use the `kedro viz deploy` command to publish Kedro-**V**iz on **GCP**.

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
````diff
@@ -0,0 +1,115 @@
+Use the `kedro viz deploy` command to publish Kedro-viz on Azure. You can execute the following command from your project's root folder:
+
+```bash
+kedro viz deploy --platform=azure --endpoint=[azure-endpoint] --bucket-name=[azure-bucket-name]
````
```bash
kedro viz deploy --platform=gcp --endpoint=[gcp-endpoint] --bucket-name=[gcp-bucket-name]
```

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
```diff
@@ -0,0 +1,115 @@
+
+Kedro-Viz does not handle billing. You pay for storing objects on your Google Cloud Storage. The amount you pay depends on the amount of data stored, data processing and network usage. Additionally you may be charged for using cloud load balancing.
+
+See the official [Google Cloud Storage Billing](https://cloud.google.com/storage/pricing) and [Google Cloud Load Balancing Billing](https://cloud.google.com/vpc/network-pricing#lb) for more information.
```
[Nit] and [Google Cloud Load **Balancer** Billing]

ravi-kumar-pilla commented:
Hi Rashida, The docs look crisp and really nice to follow. I have left some comments but most of them are non-blocking for release. Great work !!

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
```diff
@@ -0,0 +1,114 @@
+
+### Publish and share via CLI
+
+Use the `kedro viz deploy` command to publish Kedro-Viz on Azure. You can execute the following command from your project's root folder:
```
on GCP

ravi-kumar-pilla commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
```diff
@@ -0,0 +1,114 @@
+# Publish and share on GCP
+
+This page describes how to publish Kedro-Viz on Azure to share it with others. It uses the spaceflights tutorial as an example.
```
on GCP

astrojuanlu commented on docs/source/platform_agnostic_sharing_with_kedro_viz.md:
````diff
@@ -0,0 +1,47 @@
+
+```text
+fsspec>=2023.9.0
+kedro>=0.18.2
````
0.19.1? (for consistency with the instructions)

astrojuanlu commented on docs/source/platform_agnostic_sharing_with_kedro_viz.md:
```diff
@@ -0,0 +1,47 @@
+
+## Static website hosting platforms such as GitHub Pages
+
+Follow the steps [listed in the GitHub pages documentation](https://docs.github.com/en/pages/quickstart) to create a Git repository that supports GitHub Pages. On completion, push the contents of the previously created `build` folder to this new repository. . Your site will be available at the following URL: `http://<username>.github.io`
```
Double dot

astrojuanlu commented on docs/source/publish_and_share_kedro_viz_on_aws.md:
````diff
@@ -0,0 +1,119 @@
+
+```text
+fsspec>=2023.9.0
+kedro>=0.18.2
````
0.19.1?

rashidakanchwala replied:
that's true but then we are implying shareable viz doesn't work with Kedro 18 which is not true and it does. 

i think the 0.19.1 from earlier was just that to get the best and latest be on the latest kedro version.

astrojuanlu commented on docs/source/publish_and_share_kedro_viz_on_gcp.md:
````diff
@@ -0,0 +1,114 @@
+
+```text
+fsspec>=2023.9.0
+kedro>=0.18.2
````
0.19.1?
